kale/task.py

```diff
@@ -295,3 +355,57 @@ def _report_permanent_failure(cls, message, exception, failure_type,
                        (failure_type, message.task_name, message.task_id,
                         exception))
         logger.error(message_str)
+
+
+class RemoteException(object):
+    """Utility class for reraising an exception from a remote process.
+
+    Keeps the original stack trace.
+    """
+    def __init__(self):
+        self.exc_info = sys.exc_info()
+
+    def reraise(self):
+        six.reraise(*self.exc_info)
+
+
+def _task_wrapper(task_inst, args, kwargs):
+    """Task wrapper function.
+
+    In single-process mode, this will return any exception raised.
+
+    In multi-process mode, this will return a RemoteException for any exceptions raised.
+    """
+    # This enables tasks to be blacklisted by ID or type.
+    task_inst._setup_task_environment()
+    task_inst._pre_run(*args, **kwargs)
+
+    if settings.is_multiprocess_mode():
+        # Remove SIGTERM signal handler; the parent process handles all the cleanup.
+        signal.signal(signal.SIGTERM, signal.SIG_DFL)
+
+    try:
+        # This raises an exception if the task should not be attempted.
+        task_inst._check_blacklist(*args, **kwargs)
+        task_inst.run_task(*args, **kwargs)
+    except Exception as exc:
+        # Cleanup the environment this task was running in right away.
+        task_inst._clean_task_environment(task_id=task_inst.task_id,
+                                          task_name=task_inst.task_name, exc=exc)
+        if settings.is_multiprocess_mode():
+            return RemoteException()
+        return exc
```

Comment:
That's what the RemoteException & tblib handles! It pickles the traceback and it gets reraised with the stack trace from the task. I fully understand this requirement and its met. The stack trace will be a little different, since it'll start from _task_wrapper, but it'll contain the full stack from the task.